src/Microsoft.Unity.Analyzers/UnusedCoroutineReturnValue.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Unity.Analyzers.Resources;
+
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class UnusedCoroutineReturnValueAnalyzer : DiagnosticAnalyzer
+	{
+		public const string Id = "UNT0012";
+
+		private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			Id,
+			title: Strings.UnusedCoroutineReturnValueDiagnosticTitle,
+			messageFormat: Strings.UnusedCoroutineReturnValueDiagnosticMessageFormat,
+			category: DiagnosticCategory.Correctness,
+			defaultSeverity: DiagnosticSeverity.Info,
+			isEnabledByDefault: true,
+			description: Strings.UnusedCoroutineReturnValueDiagnosticDescription);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		private static readonly Type coroutineReturnValue = typeof(System.Collections.IEnumerable);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.RegisterSyntaxNodeAction(AnalyzeInvocation, SyntaxKind.InvocationExpression);
+		}
+
+		private static void AnalyzeInvocation(SyntaxNodeAnalysisContext context)
+		{
+			var typeInfo = context.Compilation.GetTypeByMetadataName("System.Collections.IEnumerator");
```

Comment:
nameof(System.Collections.IEnumerator) instead ?

Reply:
`typeof(System.Collections.IEnumerator).FullName`, nameof only returns the typename, not the fully qualified name.

Reply:
> `typeof(System.Collections.IEnumerator).FullName`, nameof only returns the typename, not the fully qualified name.

Yep saw that after running tests :)